### PR TITLE
multi-asset tests for updateRecords

### PR DESCRIPTION
**Pull request by hieuvubk**
Closes: #2688

## What is the purpose of the change

> Update more tests for twap/updateRecords( multi-asset case)

- creates new records
- with errs

### CODE REVIEW

File: x/twap/logic_test.go [hunk 1]
```diff
@@ -1044,9 +1043,85 @@ func (s *TestSuite) TestUpdateRecords() {
 				},
 			},
 		},
-		// TODO: complete multi-asset pool tests:
-		// "multi-asset pool; pre-set at t and t + 1; creates new records": {},
-		// "multi-asset pool; pre-set at t and t + 1; pre-existing records some with error and some with too large spot price, overwrites erorr time":                        {},
+		"multi-asset pool; pre-set at t and t + 1; creates new records": {
+			preSetRecords: []types.TwapRecord{threeAssetRecordAB, tPlus10sp5ThreeAssetRecordAB},
```
Comment (p0mvn) [inline x/twap/logic_test.go:1047]: For multi-asset pools, there should be at least 3 records at a given time.

Assume a 3 asset pool with assets A, B and C.

Then, if swap occurs at time t, we should have AB, AC, and BC records (for each pair).

Given that this test is meant for time t and t + 1, we need 6 records.

We should also have 3 spot price overrides and 3 most recent records.

**Review by p0mvn — CHANGES_REQUESTED** [review]

File: x/twap/logic_test.go (hunk 1 above)
Comment (hieuvubk) [reply x/twap/logic_test.go]: Ax thank you! Already updated could u pls check

File: x/twap/logic_test.go [hunk 2]
```diff
@@ -1047,9 +1047,190 @@ func (s *TestSuite) TestUpdateRecords() {
...
+				// The original record BC at t + 1.
+				{
+					spotPriceA:    tPlus10sp5ThreeAssetRecordBC.P0LastSpotPrice,
+					spotPriceB:    tPlus10sp5ThreeAssetRecordBC.P1LastSpotPrice,
+				},
+				// The new record BC added.
+				{
+					spotPriceA:    sdk.OneDec(),
+					spotPriceB:    sdk.OneDec().Add(sdk.OneDec()),                            
+					isMostRecent:  true,
```
Comment (mattverse) [inline x/twap/logic_test.go:1230]: should have `lastErrorTime`?

File: x/twap/logic_test.go [hunk 3]
```diff
@@ -1047,9 +1047,190 @@ func (s *TestSuite) TestUpdateRecords() {
...
+				},
+				// The new record BC added.
+				{
+					spotPriceA:   sdk.OneDec().Add(sdk.OneDec()),
+					spotPriceB:   sdk.OneDec(),
+					isMostRecent: true,
+				},
+			},
+		},
+		"multi-asset pool; pre-set at t and t + 1 with err, large spot price, overwrites erorr time": {
```
Comment (mattverse) [inline x/twap/logic_test.go:1140]:
```suggestion
		"multi-asset pool; pre-set at t and t + 1 with err, large spot price, overwrites error time": {
```

**Review by mattverse — COMMENTED** [review]
The test cases and the comments lgtm! Just need a question answered on adding lastErrorTime for one of the expected test cases and should be 🌮

**Review by p0mvn — APPROVED** [review]
LGTM. Nice work

**Review by mattverse — APPROVED** [review]

**Comment by mattverse** [comment]
Removed unnecessary whitelines within the test cases, merging this once final CI check has finished!

**Pull request merged**